enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,373 @@
+architecture appropriate values. The API has to make a provision to include the architecture value so the AMI and instance type can be chosen
+accordingly.
+
+### Console
+
+- As of today console filters content based on control plane architecture. This needs to be entirely removed or made configurable for
+heterogeneous clusters. The operator should be able to filter out the set of OLM operators that, if installed will be able to find
+worker nodes to run on. This means that it has to display only operators that can be supported by architectures that the workers are
+running on.
+- Console would provide a way for users to select which architecture nodes they would potentially want to deploy the operator on. This would be
+filtered to provide a list of architectures that the operator supports.
+
+### Install
+
+For Phase 1, the way to add more workers to the cluster is to handcraft a machineset manifest file for a worker node and specify the architecture
+specific details like instancetype, image references and deploy the machineset to get a worker node. This can be pseudo automated further by creating
+the manifests using `openshift-install create manifests` and modifying or adding worker manifest files. Eventually the installer should support being
+able to declare differing compute and control plane sections in the install-config mainly allowing architectures to be different. In doing so it would
+also generate appropriate machineset and machineconfig files tailored to the specific architecture. The machineset file would contain the instancetype,
+image id (AMI or Azure disk image etc..) and region corresponding to the architecture chosen for the workers.
+
+### Open Questions
+
+- What would be the method to provide recommendation updates when moving from a homogeneous graph to a heterogeneous graph ?
+
+### Test Plan
+
+Testing for Phase 1:
+- QE would need to test heterogeneous nightly payloads before the entry is added to Cincinnati. This implies that the heterogeneous payloads will be
+added to the candidate channel a bit later than the homogeneous payloads.
+- QE would test heterogeneous payloads on Azure and AWS platforms with the cluster configuration mainly being x86 control plane and ARM worker nodes.
+- Testing scenarios would include machine set scaling, upgrades involving ARM nodes and OVN testing on heterogeneous clusters.
+- New release payload acceptance jobs on heterogeneous clusters.
+
+Testing for Phase 2:
+- With the addition of Power and Z architectures, there will be more permutations to test. We would need to figure out the absolute essential set of
+permutations and the tests to run on those.
+- Imagestream conformance tests.
+- OLM and console testing
+- Test various upgrade scenarios (homogeneous -> heterogeneous and vice versa)
+
+### Graduation Criteria
+
+Phase 1 will have a pre-4.11 dev preview and 4.11 tech preview with documented limitations.
```

Comment:
Do we set the techpreview no upgrade feature gate? In both cases?

Reply:
yes. updated to reflect this and removed the dev preview.